server/core/src/main/java/org/infinispan/server/core/transport/NonRecursiveEventLoopGroup.java

```diff
@@ -0,0 +1,146 @@
+package org.infinispan.server.core.transport;
+
+import java.lang.invoke.MethodHandles;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.infinispan.commons.CacheException;
+import org.infinispan.commons.logging.LogFactory;
+import org.infinispan.factories.annotations.Stop;
+import org.infinispan.factories.scopes.Scope;
+import org.infinispan.factories.scopes.Scopes;
+import org.infinispan.server.core.logging.Log;
+import org.infinispan.server.core.utils.DelegatingEventLoopGroup;
+
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.MultithreadEventLoopGroup;
+import io.netty.util.concurrent.EventExecutor;
+import io.netty.util.concurrent.Future;
+import io.netty.util.internal.ThreadExecutorMap;
+
+/**
+ * This event loop group prevents any tasks submitted via {@link java.util.concurrent.ExecutorService} API from
```

Comment:
I guess `invokeAll` and `invokeAny` from `ExecutorService` are missing

Reply:
I didn't think of it because netty doesn't implement these methods in a thread pool way imo :( They just send all tasks to a single event loop. I would say we can just change the doc that only submit and execute are affected I guess.